components/insight/SRC/org/openmicroscopy/shoola/env/data/views/calls/ThumbnailLoader.java

```diff
@@ -62,395 +59,305 @@
+            sizeX = pxd.getSizeX();
+            sizeY = pxd.getSizeY();
+        } else {
+            Dimension d = Factory.computeThumbnailSize(sizeX, sizeY,
+                    pxd.getSizeX(), pxd.getSizeY());
+            sizeX = d.width;
+            sizeY = d.height;
+        }
+
+        if (userId >= 0) {
+            long rndDefId = service.getRenderingDef(ctx,
+                    pxd.getId(), userId);
+            // the user might not have own rendering settings
+            // for this image
+            if (rndDefId >= 0)
+                store.setRenderingDefId(rndDefId);
+        }
+
+        if (VersionCompare.compare(context.getGateway().getServerVersion(), VERSION_THUMBNAIL_NO_DEFAULT) >= 0) {
+            // If the client is connecting to a server with version 5.4.8 or greater, use the thumbnail
+            // loading function that doesn't return a clock.
+            return store.getThumbnailWithoutDefault(omero.rtypes.rint(sizeX),
+                    omero.rtypes.rint(sizeY));
+        } else {
+            return store.getThumbnail(omero.rtypes.rint(sizeX),
+                    omero.rtypes.rint(sizeY));
+        }
     }
 
     /**
-     * Creates a new instance.
-     * If bad arguments are passed, we throw a runtime exception so to fail
-     * early and in the caller's thread.
-     * 
-     * @param ctx The security context.
-     * @param image The {@link ImageData}, the thumbnail
-     * @param maxWidth The maximum acceptable width of the thumbnails.
-     * @param maxHeight The maximum acceptable height of the thumbnails.
-     * @param userIDs The users the thumbnail are for.
+     * Returns whether a pyramid should be used for the given {@link PixelsData}.
+     * This usually implies that this is a "Big image" and therefore will
+     * need tiling.
+     *
+     * @param pxd
+     * @return
      */
-    public ThumbnailLoader(SecurityContext ctx, ImageData image, int maxWidth,
-            int maxHeight, Set<Long> userIDs)
-    {
-        if (image == null) throw new IllegalArgumentException("No image.");
-        if (maxWidth <= 0)
-            throw new IllegalArgumentException(
-                    "Non-positive width: "+maxWidth+".");
-        if (maxHeight <= 0)
-            throw new IllegalArgumentException(
-                    "Non-positive height: "+maxHeight+".");
-        this.ctx = ctx;
-        images = new HashSet<DataObject>(1);
-        images.add(image);
-        this.maxWidth = maxWidth;
-        this.maxHeight = maxHeight;
-        this.userIDs = userIDs;
-        asImage = false;
-        service = context.getImageService();
+    private boolean requiresPixelsPyramid(PixelsData pxd) throws ServerError, DSOutOfServiceException {
+        int maxWidth = Integer.parseInt(getConfigService()
+                .getConfigValue("omero.pixeldata.max_plane_width"));
+        int maxHeight = Integer.parseInt(getConfigService()
+                .getConfigValue("omero.pixeldata.max_plane_height"));
```

Comment:
I think these two properties should go into the connect method as well and get stored as 'LookupName' so that they are only requested once.